benchmark/worker/benchmark_client.go

```diff
@@ -199,13 +200,17 @@ func startBenchmarkClient(config *testpb.ClientConfig) (*benchmarkClient, error)
 		return nil, err
 	}
 
+	rpcCountPerConn := int(config.OutstandingRpcsPerChannel)
 	bc := &benchmarkClient{
-		histogram: stats.NewHistogram(stats.HistogramOptions{
+		histogramOptions: stats.HistogramOptions{
 			NumBuckets:     int(math.Log(config.HistogramParams.MaxPossible)/math.Log(1+config.HistogramParams.Resolution)) + 1,
 			GrowthFactor:   config.HistogramParams.Resolution,
 			BaseBucketSize: (1 + config.HistogramParams.Resolution),
 			MinValue:       0,
-		}),
+		},
+		mutexes:    make([]*sync.Mutex, rpcCountPerConn*len(conns)),
+		histograms: make([]*stats.Histogram, rpcCountPerConn*len(conns)),
```

Comment:
make([]lockingHistogram, rpcCountPerConn*len(conns), rpcCountPerConn*len(conns))

specify the cap to eliminate unnecessary memory footprint.